sources/packages/backend/libs/utilities/src/compressed-file-utils.ts

```diff
@@ -0,0 +1,28 @@
+import * as AdmZip from "adm-zip";
+
+/**
+ * Reads the first extracted file from a compressed archive file.
+ * @param compressedFileBuffer
+ * @param options options.
+ * - `encoding`: encoding to read the file.
+ * @returns first extracted file name and data.
+ */
+export function readFirstExtractedFile(
+  compressedFileBuffer: Buffer,
+  options?: { encoding: string },
+): {
+  fileName: string;
+  data: string;
+} {
+  const zipFile = new AdmZip(compressedFileBuffer);
+  const [firstExtractedFile] = zipFile.getEntries();
+  if (!firstExtractedFile) {
+    throw new Error("No files found in zip file");
+  }
+  // Read the first extracted file with the specified encoding.
+  const data = zipFile.readAsText(firstExtractedFile, options?.encoding);
```

Comment:
Since there is an `async` method available (`zipFile.readAsTextAsync`), was there any reason to not use it?

Reply:
To use `readAsTextAsync` in a way to wait that callback is completed, we must promisify it. As we do not have an opportunity to run the reading in parallel with any other operation, I did not use the async option.